includes/multi-currency/Analytics.php

```diff
@@ -332,6 +333,32 @@ private function generate_case_when( string $variable, string $then, string $els
 		return "CASE WHEN {$variable} IS NOT NULL THEN {$then} ELSE {$else} END";
 	}
 
+	/**
+	 * Perform an SQL query to determine whether Multi Currency has ever been used on this store,
+	 * by checking how many orders are in the database where an exchange currency rate has been stored.
+	 *
+	 * @return bool
+	 */
+	private function has_multi_currency_orders() {
+		global $wpdb;
+
+		$orders = intval(
+			$wpdb->get_var(
+				"
+				SELECT
+					COUNT(DISTINCT orders.ID)
+				FROM
+					{$wpdb->posts} AS orders
+				INNER JOIN
+					{$wpdb->postmeta} order_meta ON order_meta.post_id = orders.ID
+					AND order_meta.meta_key = '_wcpay_multi_currency_order_exchange_rate';
```

Comment:
Not a big improvement, but thinking on squeezing every ms, maybe we could avoid the `JOIN` and just look for the meta. In the end we only want to know if there are orders with exchange rate.

```suggestion
				SELECT
					COUNT(post_id)
				FROM
					{$wpdb->postmeta}
				WHERE
					meta_key = '_wcpay_multi_currency_order_exchange_rate';
```

Reply:
Good idea! That would achieve the same result in a more efficient way. Committed the suggestion!